File: springboot-service/src/main/java/internal/api/springbootservice/entity/Message.java
package internal.api.springbootservice.entity;

import internal.api.springbootservice.payload.MessageRequest;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity(name = "message")
@Table(name = "messages")
@Getter
@Setter
@NoArgsConstructor
public class Message {

    @EmbeddedId
    private MessageId messageId;

    @ManyToOne(fetch = FetchType.LAZY)
    @MapsId(value = "name")
    @JoinColumn(name = "conversation_name")
    private Conversation conversation;

    @Column(name = "timestamp_received")
    private LocalDateTime timestampReceived;

    @Column(name = "timestamp_sent")
    private LocalDateTime timestampSent;

    public Message(Conversation conversation, long lastMessage,
                   LocalDateTime sentAt, LocalDateTime receivedAt) {
        this.messageId = new MessageId(conversation, lastMessage);
        this.conversation = conversation;
        this.timestampSent = sentAt;
        this.timestampReceived = receivedAt;
    }
}

File: dotnet-service/dotnet-service/src/Entities/Message.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace dotnet_service.Entities;

public class Message
{
    public string ConversationName { get; set; }
    
    public long MessageId { get; set; }
    
    public Conversation Conversation { get; set; }
    
    public string TimestampSent { get; set; }
    
    public string TimestampReceived { get; set; }

    public Message()
    {
        
    }
    
    public Message(long messageId, Conversation conversation, string sentAt, string receivedAt)
    {
        MessageId = messageId;
        Conversation = conversation;
        ConversationName = conversation.Name;
        TimestampSent = sentAt;
        TimestampReceived = receivedAt;
    }
}